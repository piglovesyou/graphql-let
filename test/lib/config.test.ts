import { join as pathJoin } from 'path';
import loadConfig, { loadConfigSync } from '../../src/lib/config';

const cwd = pathJoin(__dirname, '../__fixtures/config');

describe('config.ts', () => {
  beforeAll(() => {
    process.env.GRAPHQL_SERVER_ENDPOINT = 'https://yeah/graphql';
    process.env.GRAPHQL_SERVER_TOKEN = 'blaa';
  });

  afterAll(() => {
    delete process.env.GRAPHQL_SERVER_ENDPOINT;
    delete process.env.GRAPHQL_SERVER_TOKEN;
  });

  test('interpolates environment variables', () => {
    const [{ schema }] = loadConfigSync(cwd);
    expect(schema).toMatchSnapshot();
  });

  test('interpolates environment variables asynchronously', async () => {
    const [{ schema }] = await loadConfig(cwd);
    expect(schema).toMatchSnapshot();
  });
});
