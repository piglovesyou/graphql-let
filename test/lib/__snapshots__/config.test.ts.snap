// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config.ts interpolates environment variables 1`] = `
Array [
  Object {
    "https://yeah/graphql": Object {
      "headers": Object {
        "Authorization": "blaa",
      },
    },
  },
]
`;

exports[`config.ts interpolates environment variables asynchronously 1`] = `
Array [
  Object {
    "https://yeah/graphql": Object {
      "headers": Object {
        "Authorization": "blaa",
      },
    },
  },
]
`;
